packages/provider/src/api/ja4.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright 2021-2025 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createHash } from "node:crypto";
+import type { IncomingHttpHeaders } from "node:http";
+import { Readable } from "node:stream";
+import { type Logger, getLoggerDefault } from "@prosopo/common";
+import { readTlsClientHello } from "read-tls-client-hello";
+
+export const getJA4 = async (headers: IncomingHttpHeaders, logger?: Logger) => {
+	logger = logger || getLoggerDefault();
+
+	// Validate headers
+	if (
+		!headers["x-tls-clienthello"] ||
+		!headers["x-tls-version"] ||
+		!headers["x-tls-server-name"]
+	) {
+		logger.error("Missing required headers.");
+		return { error: "Missing required headers." };
+	}
+
+	// make sure the client hello message is a string
+	if (
+		typeof headers["x-tls-clienthello"] !== "string" ||
```

Comment:
see comments on provider mock version

Reply:
I've fixe this in the provider code. Provider-mock API stuff is just for us testing without having to run a provider.